lightning/src/chain/package.rs

```diff
@@ -97,30 +97,36 @@ pub(crate) struct RevokedOutput {
 	weight: u64,
 	amount: u64,
 	on_counterparty_tx_csv: u16,
+	opt_anchors: Option<()>,
+	is_counterparty_balance: Option<()>,
 }
 
 impl RevokedOutput {
-	pub(crate) fn build(per_commitment_point: PublicKey, counterparty_delayed_payment_base_key: PublicKey, counterparty_htlc_base_key: PublicKey, per_commitment_key: SecretKey, amount: u64, on_counterparty_tx_csv: u16) -> Self {
+	pub(crate) fn build(per_commitment_point: PublicKey, counterparty_delayed_payment_base_key: PublicKey, counterparty_htlc_base_key: PublicKey, per_commitment_key: SecretKey, amount: u64, on_counterparty_tx_csv: u16, opt_anchors: bool, is_counterparty_balance: bool) -> Self {
 		RevokedOutput {
 			per_commitment_point,
 			counterparty_delayed_payment_base_key,
 			counterparty_htlc_base_key,
 			per_commitment_key,
 			weight: WEIGHT_REVOKED_OUTPUT,
 			amount,
-			on_counterparty_tx_csv
+			on_counterparty_tx_csv,
+			opt_anchors: if opt_anchors { Some(()) } else { None },
+			is_counterparty_balance: if is_counterparty_balance { Some(()) } else { None },
 		}
 	}
 }
 
 impl_writeable_tlv_based!(RevokedOutput, {
 	(0, per_commitment_point, required),
+	(1, is_counterparty_balance, option),
```

Comment:
Renamed to `is_counterparty_balance_or_non_anchors`. If you're thinking about further functional changes in the way this PR is altering the mapping of malleability flags in function of package outputs, based on `opt_anchors` let me know. I don't know how `is_counterparty_balance_or_non_anchors` being always `Some()` in case of `opt_anchors` being `Some()` can solve current compatibility issues.